Hi @illume ,

This storybook snapshot test failure is unrelated to this PR, as this branch only contains changes to the utility files (`util.ts` and `util.test.ts`). 

```js
⎯⎯⎯⎯⎯⎯⎯ Failed Tests 3 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/storybook.test.tsx > Storybook Tests > Job/JobDetailsView > Default
Error: Snapshot `Storybook Tests > Job/JobDetailsView > Default 1` mismatched

- Expected
+ Received

@@ -688,11 +688,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11
    243|           document.body.removeAttribute('style');
    244| 
    245|           await expect(document.body).toMatchFileSnapshot(snapshotPath…
       |           ^
    246|         });
    247|       });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/3]⎯

 FAIL  src/storybook.test.tsx > Storybook Tests > Job/JobDetailsView > Running
Error: Snapshot `Storybook Tests > Job/JobDetailsView > Running 1` mismatched

- Expected
+ Received

@@ -593,11 +593,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11
    243|           document.body.removeAttribute('style');
    244| 
    245|           await expect(document.body).toMatchFileSnapshot(snapshotPath…
       |           ^
    246|         });
    247|       });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/3]⎯

 FAIL  src/storybook.test.tsx > Storybook Tests > VolumeAttributesClass/DetailsView > Base
Error: Snapshot `Storybook Tests > VolumeAttributesClass/DetailsView > Base 1` mismatched

- Expected
+ Received

@@ -218,11 +218,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11
    243|           document.body.removeAttribute('style');
    244| 
    245|           await expect(document.body).toMatchFileSnapshot(snapshotPath…
       |           ^
    246|         });
    247|       });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/3]⎯


  Snapshots  3 failed
 Test Files  1 failed | 122 passed (123)
      Tests  3 failed | 1775 passed (1778)
   Start at  14:45:00
   Duration  92.40s (transform 18.32s, setup 6.85s, collect 113.74s, tests 76.14s, environment 32.69s, prepare 7.69s)


Error: Error: Snapshot `Storybook Tests > Job/JobDetailsView > Default 1` mismatched

- Expected
+ Received

@@ -688,11 +688,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11



Error: Error: Snapshot `Storybook Tests > Job/JobDetailsView > Running 1` mismatched

- Expected
+ Received

@@ -593,11 +593,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11



Error: Error: Snapshot `Storybook Tests > VolumeAttributesClass/DetailsView > Base 1` mismatched

- Expected
+ Received

@@ -218,11 +218,23 @@
                    >
                      Events
                    </h2>
                    <div
                      class="MuiBox-root css-ldp2l3"
-                   />
+                   >
+                     <button
+                       aria-label="Events lifetime information"
+                       class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-ai5t7w-MuiButtonBase-root-MuiIconButton-root"
+                       data-mui-internal-clone-element="true"
+                       tabindex="0"
+                       type="button"
+                     >
+                       <span
+                         class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+                       />
+                     </button>
+                   </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1txv3mw"

 ❯ src/storybook.test.tsx:245:11


make: *** [Makefile:353: frontend-test] Error 1
```

--- a/frontend/src/lib/util.test.ts
+++ b/frontend/src/lib/util.test.ts
@@ -16,6 +16,7 @@
 
 import {
   combineClusterListErrors,
+  compareUnits,
   flattenClusterListItems,
   formatDuration,
   getPercentStr,
@@ -342,5 +343,46 @@ describe('normalizeUnit', () => {
     it('shows plural cores', () => {
       expect(normalizeUnit('cpu', '2')).toBe('2 cores');
     });
+  });
+});
+
+describe('compareUnits', () => {
+  it('compares identical values', () => {
+    expect(compareUnits('500m', '500m')).toBe(true);
+  });
+
+  it('ignores whitespace differences', () => {
+    expect(compareUnits('  500 m  ', '500m')).toBe(true);
+    expect(compareUnits('500m', ' 500m ')).toBe(true);
+  });
+
+  it('ignores case differences', () => {
+    expect(compareUnits('500Mi', '500mi')).toBe(true);
+    expect(compareUnits('2ki', '2Ki')).toBe(true);
+  });
+
+  it('treats normalized formatting as equal when numeric value matches', () => {
+    expect(compareUnits('1k', normalizeUnit('memory', '1k'))).toBe(true);
+  });
+
+  it('returns true for matching decimal/fractional numbers', () => {
+    expect(compareUnits('1.5', '1.50')).toBe(true);
+    expect(compareUnits('2.3Mi', '2.3')).toBe(true);
+  });
+
+  it('returns false for mismatched decimal/fractional values', () => {
+    expect(compareUnits('1.5Gi', '1.61 GB')).toBe(false);
+    expect(compareUnits('2.3', '2.4')).toBe(false);
+  });
+
+  it('returns false for mismatched parsed values', () => {
+    expect(compareUnits('500m', '250m')).toBe(false);
+    expect(compareUnits('1Ki', '2Ki')).toBe(false);
+  });
+
+  it('returns false when either value does not start with a number (NaN cases)', () => {
+    expect(compareUnits('abc', '1')).toBe(false);
+    expect(compareUnits('1', 'abc')).toBe(false);
+    expect(compareUnits('', '')).toBe(false);
   });
 });

--- a/frontend/src/lib/util.ts
+++ b/frontend/src/lib/util.ts
@@ -470,14 +470,16 @@ export function useURLState<T extends string | number | undefined = string>(
   return [value, setValue] as [T, React.Dispatch<React.SetStateAction<T>>];
 }
 
-// compareUnits compares two units and returns true if they are equal
+// compareUnits compares the parsed numeric portion of two quantities (ignoring unit suffixes) and returns true if they are equal.
 export function compareUnits(quantity1: string, quantity2: string) {
   // strip whitespace and convert to lowercase
   const qty1 = quantity1.replace(/\s/g, '').toLowerCase();
   const qty2 = quantity2.replace(/\s/g, '').toLowerCase();
 
-  // compare numbers
-  return parseInt(qty1) === parseInt(qty2);
+  // Compare the parsed numeric portion only (unit suffix is ignored).
+  const n1 = Number.parseFloat(qty1);
+  const n2 = Number.parseFloat(qty2);
+  return !Number.isNaN(n1) && !Number.isNaN(n2) && n1 === n2;
 }
 
 export function normalizeUnit(resourceType: string, quantity: string) {

